ui/imports/shared/views/HistoryView.qml

```diff
@@ -246,4 +251,45 @@ ColumnLayout {
             }
         }
     }
+
+    Component {
+        id: headerComp
+        Item {
+            visible: !d.isLoading && transactionListRoot.count !== 0
+            width: root.width
+            height: filterButton.height + 11
+            StatusRoundButton {
+                id: filterButton
+                anchors.top: parent.top
+                anchors.left: parent.left
+                width: 32
+                height: 32
+                type: StatusFlatRoundButton.Type.Primary
+                icon.name: "filter"
+                onClicked: activityFilter.popup(filterButton.x, filterButton.height)
+            }
+        }
+    }
+
+    ActivityFilterMenu {
+        id: activityFilter
+        selectedTime: ActivityFilterMenu.All
+        onSetSelectedTime: {
+            // To do connect with n=backend to set time range
+            if(selectedTime === ActivityFilterMenu.Custom) {
+                customDateRangePicker.open()
+            }
+        }
+    }
+
+    // To-do update once https://github.com/status-im/status-desktop/pull/10916 is updated and connect with backend values
+    StatusDateRangePicker {
+        id: customDateRangePicker
+        width: 440
+        height: 300
+        fromTimestamp: new Date().setDate(new Date().getDate() - 7)
+        toTimestamp: Date.now()
+        supportedStartYear: 1900
+//        onNewRangeSet: d.setCustomTimeRange(fromTimestamp, toTimestamp)
```

Comment:
You can now rebase onto latest master, it has the new `StatusDateRangePicker`

Reply:
Doing that :)

Reply:
Well, you could've kept the `d.setCustomTimeRange()` function and do nothing there ;)